File: cpp/csvviewer/csvviewer.tests/ui.tests.cpp
#include <gmock/gmock.h>
#include "Ui.h"

namespace csvviewer
{
    using namespace testing;
    TEST(UiTests, Formatting_very_simpel)
    {
        auto result = Ui::CreateLines({{{"a", "b", "c"}}, {{"1", "2", "3"}}});

        EXPECT_THAT(result, Eq(std::vector<std::string>{"a|b|c", "-+-+-", "1|2|3"}));
    }

    TEST(UiTests, Formatting_of_different_widths)
    {
        auto result = Ui::CreateLines({{{"a", "bb", "ccc"}}, {{"111", "22", "3"}}});

        EXPECT_THAT(result, Eq(std::vector<std::string>{"a  |bb|ccc", "---+--+---", "111|22|3  "}));
    }
}

File: csharp/csvviewer/csvviewer/csvviewer.tests/UiTests.cs
using NUnit.Framework;

namespace csvviewer.tests
{
    [TestFixture]
    public class UiTests
    {
        [Test]
        public void Formatting_very_simple() {
            var sut = new Ui();
            var result = sut.CreateLines(new[] {
                new Record { Values = new[]{"a", "b", "c"}},
                new Record { Values = new[]{"1", "2", "3"}}
            });
            Assert.That(result, Is.EqualTo(new[]{"a|b|c", "1|2|3"}));
        }

        [Test]
        public void Formatting_of_different_widths() {
            var sut = new Ui();
            var result = sut.CreateLines(new[] {
                new Record { Values = new[]{"a", "bb", "ccc"}},
                new Record { Values = new[]{"111", "22", "3"}}
            });
            Assert.That(result, Is.EqualTo(new[]{"a  |bb|ccc", "111|22|3  "}));
        }
    }
}